hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/ObjectStore.java

```diff
@@ -458,6 +458,23 @@ public boolean addAcl(OzoneObj obj, OzoneAcl acl) throws IOException {
     return proxy.addAcl(obj, acl);
   }
 
+  /**
+   *
+   * @param obj Ozone object for which acl should be checked.
+   * @param acl ozone acl to be checked.
+   * @return true if acl exists in the obj, else false.
+   * @throws IOException if there is error.
+   */
+  public boolean checkAclExist(OzoneObj obj, OzoneAcl acl) throws IOException {
+    for (OzoneAcl existAcl : proxy.getAcl(obj)) {
+      if (existAcl.equals(acl)) {
```

Comment:
This logic is not completely correct, take a case like USER1:RW, and we are trying to add USER1:R, the acl already exists. equals() will not work in this case.

Reply:
I think from below comment we don't need this new method.